### PR TITLE
added phpdocs

### PR DESCRIPTION
**Pull request by clxmstaab**

### CODE REVIEW

File: src/Composer/Downloader/GitDownloader.php [hunk 1]
```diff
@@ -27,9 +27,21 @@
  */
 class GitDownloader extends VcsDownloader implements DvcsDownloaderInterface
 {
+    /**
+     * @var array<string, bool>
+     */
     private $hasStashedChanges = array();
+    /**
+     * @var array<string, bool>
```
Comment (Seldaek) [inline src/Composer/Downloader/GitDownloader.php:35]: Is this syntax supported by phpstorm & co? Or does one need to use psalm-var + `@var bool[]`?

Comment (clxmstaab) [reply src/Composer/Downloader/GitDownloader.php:35]: arghs didn't think of the IDE.. you are right... will fix

**Comment by clxmstaab** [comment]
@Seldaek fixed per feedback

**Comment by Seldaek** [comment]
Thanks, it is a PITA tbh and it'd be much nicer if there was common agreed upon syntax for this :/

**Comment by staabm** [comment]
yeah absolutely. hopefully the phpstorm team reads its bugtracker and reacts based on the issue which have hundreds of +1s on this topic

**Pull request merged**